pkg/cmd/cli-manager/cmd.go

```diff
@@ -27,6 +27,9 @@ func NewCLIManagerCommand(name string) *cobra.Command {
 	cmd.Use = name
 	cmd.Short = "Start the CLI manager controllers"
 
+	cmd.Flags().BoolVar(&ServeArtifactAsHttp, "serve-artifacts-in-http", false, "serving artifact in HTTP instead of HTTPS. That is used for testing purposes and not recommended for production")
```

Comment:
In order to download binaries, krew performs simply GET https://github.com/kubernetes-sigs/krew/blob/6e7a20d181394ce108e408b69e67f57680bcf203/internal/download/fetch.go#L40 to the server. However, there is no option to configure to use our self-signed CA bundles. Therefore, we have to introduce this field to serve our binaries in e2e tests in `http`. This flag will be hidden and will only used for e2e tests.

Reply:
If the intention is testing purposes only then "That is used for testing purposes and not recommended for production" -> "That is used for testing purposes only. Using the flag in production is at your own risk. This flag is not supported."

Reply:
Alternatively, you could build a separate binary/command used only for testing that will expose this flag. Given the e2e is used only for us (run by GH/Prow) there's no risk of doing so. The dockerfile used for production can be different from the one used for testing. As long as the new command is "just" a wrapper on top of the current one.

Reply:
> Alternatively, you could build a separate binary/command used only for testing that will expose this flag. Given the e2e is used only for us (run by GH/Prow) there's no risk of doing so. The dockerfile used for production can be different from the one used for testing. As long as the new command is "just" a wrapper on top of the current one.

That is a good idea. Because we already have a discrimination between production `Dockerfile` and CI `Dockerfile.ci` and we can expose an environment variable to enable this flag in CI (which means that CLI Manager operator should also export this env variable in it's CI tests). Thank you.

Reply:
@ingvagabund I've added new environment variable set only in `Dockerfile.ci` to expose this flag. Please let me know what you think. Thanks.